### PR TITLE
fix: Refactor ParseRequest with helper functions and conditional parsing

### PR DESCRIPTION
**Pull request by ontehfritz**
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved request parsing: enforces request-line and combined frontmatter/body size limits, returns immediately for empty payloads, handles frontmatter that ends at end-of-input, and ensures metadata is present (empty when frontmatter has only comments).
* **Tests**
  * Added tests covering oversized request-line rejection, frontmatter-at-EOF, and frontmatter-with-only-comments edge cases.

### CODE REVIEW

File: protocol/request.go [hunk 1]
```diff
@@ -27,107 +27,161 @@ const MaxRequestFrontmatterLength = 65536 // 64KB
...
+	var fmEnd, bodyStart int
+	if idx := bytes.Index(inner, []byte(frontmatterClose)); idx >= 0 {
+		fmEnd = idx
+		bodyStart = idx + len(frontmatterClose)
+	} else if bytes.HasSuffix(inner, []byte(frontmatterTrim)) {
+		fmEnd = len(inner) - len(frontmatterTrim)
+		bodyStart = len(inner)
+	} else {
+		return nil, nil, fmt.Errorf("malformed request: unclosed frontmatter")
 	}
```
[the remark this answers is not in the document]
Comment (ontehfritz) [reply protocol/request.go:160]: The canonical empty form already works** (`---\n\n---\n`). Only the pathological zero-bytes-between-fences form is rejected.
2. **Not a regression.** The original code before my refactor had the same behavior — both parsers (request and response) look for `\n---\n` between opening and closing fences, so some byte (even just `\n`) is required between them.
3. **Our writer never emits this form.** `WriteTo` only writes frontmatter when `len(Metadata) > 0`, so we never produce `---\n---\n` on the wire.
4. **Response parser is identical in this respect.** Accepting the touching-fence form in request.go but rejecting it in response.go would introduce parser asymmetry.
5. **No reported interop issue.** Caztor is read-only. No known client emits `---\n---\n`.

The lean parser here follows "be conservative in what you accept" — legitimate empty frontmatter has an established canonical form. Accepting more edge cases for strict conformance's sake widens the attack/surprise surface without buying anything concrete.

Not applying. If an actual interop case surfaces (some third-party writer emits `---\n---\n`), I'd want to fix request.go, response.go, and the spec together — that's a protocol-level consistency change, not a localized parser tweak.

**Comment by ontehfritz** [comment]
@coderabbitai I need this approved to merge

**Comment by ontehfritz** [comment]
@coderabbitai approve
